src/jit.rs

```diff
@@ -846,32 +846,27 @@ impl<'a, C: ContextObject> JitCompiler<'a, C> {
     #[inline]
     fn emit_sanitized_load_immediate(&mut self, size: OperandSize, destination: u8, value: i64) {
         match size {
-            OperandSize::S32 => {
-                let key = self.diversification_rng.gen::<i32>() as i64;
-                self.emit_ins(X86Instruction::load_immediate(size, destination, (value as i32).wrapping_sub(key as i32) as i64));
-                self.emit_ins(X86Instruction::alu(size, 0x81, 0, destination, key, None));
-            },
             OperandSize::S64 if value >= i32::MIN as i64 && value <= i32::MAX as i64 => {
                 let key = self.diversification_rng.gen::<i32>() as i64;
-                self.emit_ins(X86Instruction::load_immediate(size, destination, value.wrapping_sub(key)));
+                self.emit_ins(X86Instruction::load_immediate(OperandSize::S64, destination, value.wrapping_sub(key)));
```

Comment:
Changing the `size` parameter in these calls has no effect, because you are already matching the size to `OperandSize::S64` in the match statement. Perhaps we should leave `size` there for consistency?

Reply:
The `size` parameter was removed so it has to be explicit now.